[Feature Flags] Use cases: kill switches, dynamic config, progressive rollouts (#36920)
DataDog/documentation: content/en/feature_flags/use_cases/kill_switches.md

---
title: Kill Switches
description: Use Boolean feature flags as kill switches to disable features instantly without redeploying.
further_reading:
- link: "/feature_flags/client/"
  tag: "Documentation"
  text: "Client-Side SDKs"
---

A **kill switch** is a Boolean feature flag where the default state is **on** (`true`). The feature runs normally until you enact the kill switch by adding a targeting rule that serves `false`.

> Use a kill switch when you need to turn off risky or broken functionality in seconds from Datadog, without waiting for a deploy. During incidents, that gives operations and engineering a fast path to stop customer impact while the rest of the application keeps running.

Reviewer: To avoid misreading that Datadog's functionality is risky or broken:
```suggestion
Use a kill switch when you need to turn off risky or broken functionality by using Datadog, without waiting for a deploy. During incidents, that gives operations and engineering a fast path to stop customer impact while the rest of the application keeps running.
```
